packages/node/test/prepare.js

```diff
@@ -0,0 +1,70 @@
+// @ts-check
```

Comment:
(should) That bash oneliner was ugly, but this is complex. Let's talk about tradeoffs. I'm not sure if we're getting enough out of this to make up for having this setup and detection.

Some of it seems to be for the node 14 support. That might be going away soon enough.  The multi-pkg-manager test is a new addition, so maybe we could skip running it on node 14

Reply:
The other problem with the bash oneliner is that it is not portable.  I am all for removing unnecessarily non-portable scripts wherever possible.

Reply:
no; neither `prepare.sh` nor any embedded oneliner is portable.  what it really does is _makes it more difficult to contribute_.  even if you, me, and the rest of the team is using a POSIX OS, we can't guarantee that.

it'd also be a good idea to add windows/pwsh to our CI matrix eventually.